gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -326,6 +329,26 @@ public GsonBuilder setFieldNamingStrategy(FieldNamingStrategy fieldNamingStrateg
     return this;
   }
 
+  /**
+   * Disallows duplicate JSON object properties during deserialization. When the JSON
+   * data contains multiple properties with the same name a {@link JsonSyntaxException}
+   * will be thrown. This affects the built-in adapters for {@code Object}, {@code Map}
```

Comment:
The `Calendar` adapter is not mentioned here (despite checking for duplicates as well) because it is most likely not that commonly used and I am not sure if its JSON format is specified anywhere in the first place.